### PR TITLE
chore: migrate project commands from Make to Just 

### PR DESCRIPTION
**Pull request by alejandrodnm**
This change replaces all Makefiles with Justfiles, making it easier to
run project commands and manage tasks. Just provides a simpler way to 
organize commands and run commands, especially in a multi-project setup.

- Switch from make to just: Converted all make commands to just.
- Update Documentation: Changed instructions in DEVELOPMENT.md to use
  just instead of make.
- Add top-level justfile: allows running all just recipes from the root
  directory.

How to Use:

Run `just -l` to see all available commands, or refer to DEVELOPMENT.md
for specific instructions.

Recipes for the projects located in `projects/*/justfile` are executed
by using the project name after the `just` command. For example `just pgai clean`, `just extension clean`.

Other examples:

```bash
$ just --summary
install-commit-hook extension::build ... pgai::build ...

$ just -l
Available recipes:
    install-commit-hook # Install semantic commit message hook
    extension ...       # pgai postgres extension. List recipes with `just -l extension`
    pgai ...            # pgai python library. List recipes with `just -l pgai`

$ just -l pgai
Available recipes:
    build        # Build source distribution and wheel package
    clean        # Remove build artifacts and temporary files
    docker-build # Build Docker image with version tag
    docker-rm    # Remove the Docker container and its volumes
    ...

$ just pgai test
...
$ just -l extension
...
$ just extension docker-build
...
```

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (jgpruitt) [inline .github/workflows/ci.yml:1]: The README.md needs to be updated. The install-by-source instructions rely on Make. I don't think DBAs will be keen on having to install `just` and `sh` in order to install the extension. I think we need to change it to call `build.py` directly I guess.

**Review by jgpruitt — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -118,10 +118,10 @@ To install pgai from source on a PostgreSQL server:
    These extensions are automatically added to your PostgreSQL database when you
    [Enable the pgai extension](#enable-the-pgai-extension-in-your-database).
 
-1. Make this `pgai` extension:
+1. Install the `pgai` PostgreSQL extension:
 
     ```bash
-    make install
+    just ext install
```
Comment (jgpruitt) [inline README.md:124]: Can we add 

"or if you don't have just"
```bash
projects/extension/build.py install
```

**Review by jgpruitt — CHANGES_REQUESTED** [review]
one last nit. sorry

File: README.md (hunk 1 above)
Comment (alejandrodnm) [reply README.md:124]: Good idea. Already updated.

**Review by jgpruitt — APPROVED** [review]

**Pull request merged**